### PR TITLE
feat(GRW-1306): added base product snap component

### PR DESCRIPTION
**Pull request by guilhermespopolin**
## Describe your changes

Adds a base component for implementing _Top Picks_ insurance list for Promise Listing Page. I've decided to call it Snap because the way I see it's not exactly a carousel.

Screenshot 2022-08-02 at 13 44 47

In order to test it, please checkout the branch and verify it with storybook

image

## Jira issue(s): [GRW-1306](https://hedvig.atlassian.net/browse/GRW-1306)

### CODE REVIEW

File: apps/store/src/components/ProductCardSnap/ProductCardSnap.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,49 @@
+import { INITIAL_VIEWPORTS } from '@storybook/addon-viewport'
+import { ComponentMeta, ComponentStory } from '@storybook/react'
+import { ProductCard } from '@/components/ProductCard/ProductCard'
+import { ProductCardSnap } from './ProductCardSnap'
+
+console.log(INITIAL_VIEWPORTS)
```
Comment (robinandeer) [inline apps/store/src/components/ProductCardSnap/ProductCardSnap.stories.tsx:6]: delete ;)

[deletstar.jpg)](https://app.graphite.dev/macros/HedvigInsurance)

File: apps/store/src/components/ProductCardSnap/ProductCardSnap.tsx [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import styled from '@emotion/styled'
+import React from 'react'
```
Comment (robinandeer) [inline apps/store/src/components/ProductCardSnap/ProductCardSnap.tsx:2]: nit: import not needed with React 17

**Review by robinandeer — COMMENTED** [review]
Nice work! Haven't tested locally but I think it looks good

My only question would be if `ProductCardSnap` needs to care about what goes inside of it?

Could we just call it e.g. `Slideshow`?

File: apps/store/src/components/ProductCardSnap/ProductCardSnap.stories.tsx (hunk 1 above)
Comment (guilhermespopolin) [reply apps/store/src/components/ProductCardSnap/ProductCardSnap.stories.tsx:6]: 🤦🏻

File: apps/store/src/components/ProductCardSnap/ProductCardSnap.tsx (hunk 2 above)
Comment (guilhermespopolin) [reply apps/store/src/components/ProductCardSnap/ProductCardSnap.tsx:2]: I see. But if I remove it then it complains about the line where I call `React.Children`
`'React' refers to a UMD global, but the current file is a module. Consider adding an import instead.`

**Review by robinandeer — APPROVED** [review]
Looks great to me! The only "nit" left is to fix unify how to spell: "SlideShow" or "Slideshow" 😅 (folder vs. module name)

**Comment by guilhermespopolin** [comment]
> Looks great to me! The only "nit" left is to fix unify how to spell: "SlideShow" or "Slideshow" 😅 (folder vs. module name)

Gonna fix it

**Pull request merged**